### PR TITLE
MohammadH/ Refactor all localization components to ts

### PR DESCRIPTION
**Pull request by mohammad-rework**
Changes:

-   ...

## Type of change

-   [ ] Bug fix
-   [ ] New feature
-   [ ] Update feature
-   [x] Refactor code
-   [ ] Translation to code
-   [ ] Translation to crowdin
-   [ ] Script configuration
-   [ ] Improve performance
-   [ ] Style only
-   [ ] Dependency update
-   [ ] Documentation update
-   [ ] Release

### CODE REVIEW

File: src/components/localization/language-switcher.tsx [hunk 1]
```diff
@@ -9,11 +8,17 @@ import Dropdown from './language-dropdown'
 import { isProduction } from 'common/websocket/config'
 import { nonENLangUrlReplace } from 'common/utility'
 
+type LanguageSwitchProps = {
+    i18n?: { language: string }
+    is_high_nav?: boolean
+    security?: 'true' | 'false'
```
Comment (michio-rework) [inline src/components/localization/language-switcher.tsx:14]: this should be `boolean`, thanks :pray:

File: src/components/localization/localized-link.tsx [hunk 2]
```diff
@@ -15,7 +14,32 @@ import {
...
+    target?: string
+    type?: string
+}
+
+type LocalizedLinkProps = ExternalLinkProps & {
+    external?: boolean | 'true'
+    weight?: string
+}
+
+export const SharedLinkStyle = css<{ active: boolean; disabled: boolean }>`
```
Comment (michio-rework) [inline src/components/localization/localized-link.tsx:42]: extract a type here please, thanks :pray:

**Review by michio-rework — CHANGES_REQUESTED** [review]

File: src/pages/bug-bounty/index.tsx [hunk 3]
```diff
@@ -16,7 +16,7 @@ const meta_attributes = {
 
 const Security = () => {
     return (
-        <Layout type="security" margin_top={10}>
```
Comment (michio-rework) [inline src/pages/bug-bounty/index.tsx:19]: this seems to be needed, or this removal is not related to your changes, can you please confirm?

File: src/pages/bug-bounty/vulnerabilities-types/index.tsx [hunk 4]
```diff
@@ -72,7 +72,7 @@ const StyledLi = styled.li`
 
 const VulnerabilitiesTypes = () => {
     return (
-        <Layout type="security" margin_top={10}>
```
Comment (michio-rework) [inline src/pages/bug-bounty/vulnerabilities-types/index.tsx:75]: this seems to be needed, or this removal is not related to your changes, can you please confirm?

File: src/pages/bug-bounty/vulnerabilities-types/index.tsx [hunk 5]
```diff
@@ -90,7 +90,7 @@ const VulnerabilitiesTypes = () => {
                                 <Flex mt="40px" ai="center">
                                     <IconWrap src={item.icon} />
                                     <Header as="h2" type="subtitle-1" ml="8px">
-                                        {localize(item.title)}
```
Comment (michio-rework) [inline src/pages/bug-bounty/vulnerabilities-types/index.tsx:93]: I think we need to translate the title here, don't we?

File: src/pages/bug-bounty/index.tsx (hunk 3 above)
Comment (mohammad-rework) [reply src/pages/bug-bounty/index.tsx:19]: We added `margin-top` to the layout component whenever the type is `"security"`. Hence `margin_top` is not needed anymore.

File: src/pages/bug-bounty/vulnerabilities-types/index.tsx (hunk 4 above)
Comment (mohammad-rework) [reply src/pages/bug-bounty/vulnerabilities-types/index.tsx:75]: Same here.

File: src/pages/bug-bounty/vulnerabilities-types/index.tsx (hunk 5 above)
Comment (mohammad-rework) [reply src/pages/bug-bounty/vulnerabilities-types/index.tsx:93]: The `item.title` already has it's own `<Localized />` component in the `vulnerability_types_content` array.

**Review by michio-rework — APPROVED** [review]
